lib/cookie.js

```diff
@@ -486,6 +505,8 @@ function parse(str, strict) {
 
   // ensure a default date for sorting:
   c.creation = new Date();
+  //NOTE: high-resolution time for the sorting in runtime
+  c.hrCreation = hrTimeNow();
```

Comment:
Can you please explain what the goal is here? I can see what it does, but I don't understand the motivation or why it's important? Since it's a small thing, perhaps you could split this off into another PR?

My concern is that if some code manually assigns to `.creation`, there's no setter that changes `.hrCreation` to match. This would result in inconsistent sorting in `cookieCompare`.